pkg/azuredisk/azuredisk.go

```diff
@@ -249,6 +249,7 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
 	if len(options.AvailabilityZone) > 0 {
 		zoneList := []string{d.cloud.GetZoneID(options.AvailabilityZone)}
```

Comment:
Got the following error when running E2E tests on a single-az cluster. The e2e test didn't catch probably because of the difference in Kubernetes version (1.15 vs 1.16):
```
1 azuredisk.go:252] azureDisk - AvailabilityZone: 0, zone list: []
...
ProvisioningFailed: failed to provision volume with StorageClass "azuredisk-7966-disk.csi.azure.com-dynamic-sc-skg8j": rpc error: code = Unknown desc = compute.DisksClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidAvailabilityZone" Message="The zone(s) '' for resource 'Microsoft.Compute/disks/pvc-01484a41-6946-416c-baae-da34ad079b9b' is not supported. The supported zones for location 'eastus2' are '1,2,3'"
```

I think we need to validate zoneID before assigning it to zoneList:

```suggestion
		zoneID := d.cloud.GetZoneID(options.AvailabilityZone)
		if len(zoneID) > 0 {
			zoneList := []string{zoneID}
			createZones = &zoneList
			klog.V(4).Infof("azureDisk - AvailabilityZone: %v, zone list: %v", options.AvailabilityZone, zoneList)
		}
```

Reply:
@chewong have you set the values of `topology.disk.csi.azure.com/zone` as `location-zonenum`, e.g. `eastus2-1` ? I got the same error if I set that value as `1`
```
allowedTopologies:
- matchLabelExpressions:
  - key: topology.disk.csi.azure.com/zone
    values:
    - eastus2-1
    - eastus2-2
```

Reply:
See https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/176/files#r342736927